test/db/postgresql/oid_types_test.rb

```diff
@@ -40,31 +34,21 @@ class SomeSample < ActiveRecord::Base
   end
 
   def test_resolves_oid_type
-    column = SomeSample.columns_hash['str']
-    if ar_version('4.2')
-      assert_instance_of ActiveRecord::Type::String, column.cast_type
-    else # 4.1/4.0
-      assert_kind_of OID::Identity, column.oid_type
-    end
+    assert_instance_of ActiveRecord::Type::String, SomeSample.type_for_attribute('str')
   end
 
   def test_returns_column_and_resolves_oid_type
     adapter = ActiveRecord::Base.connection
-    if defined? JRUBY_VERSION
-      column = adapter.column_for('some_samples', :int)
-    else
-      column = adapter.columns('some_samples').find { |c| c.name.to_sym == :int }
-    end
+    column = adapter.column_for('some_samples', :int)
     assert_not_nil column
-    assert_instance_of OID::Integer, oid_type(column)
-  end if ar_version('4.1')
+    assert_instance_of ActiveRecord::Type::Integer, column.type
+  end
 
   def test_returns_column_accessor_for_hstore
```

Comment:
@rdubya yeah we seem to have lots of perhaps too low-level tests as well which can probably go away.  I think an older motivation was to add a mixture of common tests from AR into our suite to not have to run full AR suite (or perhaps merely for regressions -- I don't actually know).  So we can start deleting stuff when it is easily verified running against Rails itself.